.github/workflows/action-codecov.yml

```diff
@@ -5,16 +5,70 @@ on: [ push, pull_request ]
 jobs:
   build:
     runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: praktikum
+          POSTGRES_PASSWORD: pass
+          POSTGRES_DB: goph_keeper
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 5s
+          --health-timeout 5s
+          --health-retries 5
+      minio:
+        # fixme: let's not depend on external unofficial image
+        image: lazybit/minio
+        ports:
+          - 9000:9000
+          - 9001:9001
+        env:
+          MINIO_ROOT_USER: praktikum
+          MINIO_ROOT_PASSWORD: pass1234567
+        options: --name=minio --health-cmd "curl http://localhost:9000/minio/health/live"
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 2
       - uses: actions/setup-go@v5
         with:
           go-version: 'stable'
+
+      - name: Install goose migrate
+        run: |
+          curl -fsSL \
+          https://raw.githubusercontent.com/pressly/goose/master/install.sh |\
+          sh
+
+      - name: Run migrations
+        run: goose -dir ./server/migrations postgres postgresql://praktikum:pass@localhost:5432/goph_keeper?sslmode=disable up
+
       - name: Run coverage
-        run: go test $(go list ./... | grep -v ./tests/)  -coverpkg=./... -race -coverprofile=coverage.out -covermode=atomic
+        run: make test
+#        run: go test $(go list ./... | grep -v ./tests/)  -coverpkg=./... -race -coverprofile=coverage.out -covermode=atomic
+#        run: go test ./...  -coverpkg=./... -race -coverprofile=coverage.out -covermode=atomic
+
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v4
         env:
-          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
\ No newline at end of file
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+
+  upload:
+    name: Generate Build Artifacts
+
+    strategy:
+      matrix:
+        os: [ macos-latest, ubuntu-latest, windows-latest ]
+        version: [ a, b, c ]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Upload
```

Comment:
This doesn't work. Every job is a new VM. You can build the project and upload it.
It should be steps, not jobs

Reply:
You need to add `go build` before the `Upload` step.